### PR TITLE
Bugfix: support set up tear down

### PR DESCRIPTION
**Pull request by alexstrat**
_tldr: I found a bug, i fixed it, I tested the fix. But tests are failing, I don't know why._

While trying to use _karma-nodeunit_ in a project, I got the feeling that `setUp` and `tearDown` did not work properly (they were considered as test-cases). Empirically, I knew that _nodeunit_ lib was buggy: changing `nodeunit.js` by a more recent build fixed the issue.

So, in _karma-nodeunit_:

I modified the tests of karma-nodeunit to make it use `setUp` and `tearDown` in d0b2dc2. As, predicted it failed test suite with a fairly explicit message:

``` sh
.
PhantomJS 1.9.2 (Mac OS X) nodeunit adapter  running a suite of tests should report test results FAILED
    Error: expected 'setUp' to equal 'Ceci nes\'t pas une test' (/Users/alex/Code/karma-nodeunit/node_modules/expect.js/expect.js:99)
PhantomJS 1.9.2 (Mac OS X) nodeunit adapter  running a suite of tests should report test results FAILED
    Error: expected 'setUp' to equal 'Ceci nes\'t pas une test' (/Users/alex/Code/karma-nodeunit/node_modules/expect.js/expect.js:99)
..
```

Then I replaced current _nodeunit_ build with a more recent one in edebd88. But things began to go weird: tests failed.

``` sh
.
PhantomJS 1.9.2 (Mac OS X) nodeunit adapter  running a suite of tests should report test results FAILED
    Error: expected false to equal true (/Users/alex/Code/karma-nodeunit/node_modules/expect.js/expect.js:99)
PhantomJS 1.9.2 (Mac OS X) nodeunit adapter  running a suite of tests should report test results FAILED
    Error: expected false to equal true (/Users/alex/Code/karma-nodeunit/node_modules/expect.js/expect.js:99)
..
```

This is saying that the "passing" fixture test-case actually does not pass.

And indeed doing:

``` diff
--- a/test/adapter.spec.js
+++ b/test/adapter.spec.js
@@ -32,7 +32,7 @@ describe('nodeunit adapter ', function() {
       sinon.stub(tc, 'result', function(result) {
         expect(result.description).to.equal("Ceci nes't pas une test");
         expect(result.suite).to.be.an('array');
-        expect(result.success).to.be(true);
+        expect(result.success).to.be(false);
         expect(result.log).to.be.an('array');
       });
       runSuite(fixtureSuite.passing, function() {
```

makes test pass.

So, to be sure that this fixture test-case was really failing, I tried to make run "normally" with `nodeunit` in NodeJS and in browser (I commited my test utility stuff in eb54d09): the fixture test-case is perfectly succeeding!

So, my guess is: whether there is a bug in adapter, wether there is a bug in its test suite. Could you help please ?

### CODE REVIEW

**Comment by hurrymaplelad** [comment]
@alexstrat another good catch.  Looks like the issue was with the test setup.  We reused the same nodeunit suite for the full suite completion test and for individual test cases.  This worked fine until setUp was introduced.  Setup wasn't called the second time the suite was run.  I suspect nodeunit marks some state on the suite.  

This situation seems unlikely outside test framework tests, so I wouldn't support it in nodeunit.  Adjusted the specs instead.  Work for you?

**Comment by alexstrat** [comment]
That worked perfectly! Thanks!
